assembly/NFT/NFT.ts

```diff
@@ -0,0 +1,226 @@
+  if (Storage.has(counterKey)) {
+    return Storage.get(counterKey);
+  } else {
+    generateEvent(`NFT not setted yet`);
+    return '';
+  }
+}
+
+/**
+ *
+ * Return the tokenId's owner
+ * @param {string} tokenId
+ * @return {string}
+ */
+export function ownerOf(tokenId: string): string {
+  if (Storage.has(ownerTokenKey + tokenId.toString())) {
+    return Storage.get(ownerTokenKey + tokenId.toString());
+  } else {
+    generateEvent(`NFT not minted yet`);
+    return '';
+  }
+}
+
+// ==================================================== //
+// ====                 TRANSFER                   ==== //
+// ==================================================== //
+/**
+ *
+ * The to address becomes the owner of the next token (if current tokenID = 10, will mint 11 )
+ * Check if max supply is not reached
+ * @param {string} args - byte string containing an owner's account (Address).
+ * @return {string}
+ */
+export function mint(args: string): string {
+  if (u32(parseInt(limitSupply(''))) > u32(parseInt(currentSupply('')))) {
+    const addr = Address.fromByteString(args);
+    _increment('');
+    const tokenID: string = currentSupply('');
+    const key = ownerTokenKey + tokenID;
+    Storage.set(key, addr.toByteString());
+    generateEvent(`tokenId ${tokenID} minted to ${addr.toByteString()} `);
+  } else {
+    generateEvent(`Max supply reached`);
+  }
+  return 'Minted';
+}
+
+/**
+ * Increment the NFT counter
+ * @param {string} _ - unused see https://github.com/massalabs/massa-sc-std/issues/18
+ * @return {string}
+ */
+function _increment(_: string): string {
+  const add1: u32 = 1;
+  const incr = u32(parseInt(Storage.get(counterKey))) + add1;
+  Storage.set(counterKey, incr.toString());
+  return '';
+}
+
+/**
+ *
+ * Return true if the caller is the creator of the SC
+ *   @param {string} _ - unused see https://github.com/massalabs/massa-sc-std/issues/18
+ *   @return {bool}
+ */
+function _onlyOwner(_: string): bool {
+  return Context.caller().toByteString() == Storage.get(ownerKey);
+}
+
+// ==================================================== //
+// ====                 TRANSFER                   ==== //
+// ==================================================== //
+
+/**
+ *Transfer a choosen token from the caller to the to Address
+ check first the caller owns the token and if token minted
+ * @param {string} args - byte string with the following format:
+ * - the recipient's account (address)
+ * - the tokenID (u64).
+ * @return {string}
+ */
+export function transfer(args: string): string {
+  const toAddress = new Address();
+  const offset = toAddress.fromStringSegment(args);
+
+  const tokenId: u64 = ByteArray.fromByteString(args.substr(offset, 8)).toU64();
+  if (!Storage.has(ownerTokenKey + tokenId.toString())) {
+    generateEvent(`token ${tokenId.toString()} not yet minted`);
+    return '';
+  }
+  if (ownerOf(tokenId.toString()) == Context.caller().toByteString()) {
```

Comment:
We can maybe create a `modifier` ( I use the Solidity term ) just as you made with the onlyOwner
But this time smth like `onlyTokenOwner` ? Will be smth that we'll use often on that contract with the allowance for instance

Reply:
done !